pkg/controller/migplan/migplan_controller.go

```diff
@@ -445,3 +470,62 @@ func (r ReconcileMigPlan) deleteImageRegistryDeploymentForClient(client k8sclien
 	}
 	return nil
 }
+
+func (r ReconcileMigPlan) ensureMigAnalytics(plan *migapi.MigPlan) error {
+	migAnalytics := &migapi.MigAnalyticList{}
+	err := r.List(context.TODO(), k8sclient.MatchingLabels(map[string]string{MigPlan: plan.Name}), migAnalytics)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return err
+		}
```

Comment:
qq: Should we not return `err` over here, like if the error is other that the `not found` one ?

Reply:
We are returning `err` in any other case except `notFound` here, and if the error is `Object not found` then we are creating that object